src/rgw/rgw_bucket_logging.cc

```diff
@@ -335,6 +344,10 @@ int rollover_logging_object(const configuration& conf,
     return ret;
   }
   if (const auto ret = target_bucket->commit_logging_object(old_obj, y, dpp); ret < 0) {
+    if (const auto ret = target_bucket->sync_owner_stats(dpp, y, nullptr); ret < 0) {
```

Comment:
we already call `quota_handler->update_stats()` as part of the bucket index transaction in `RGWRados::Object::Write::_do_write_meta()`: https://github.com/ceph/ceph/blob/210a00a/src/rgw/driver/rados/rgw_rados.cc#L3386-L3393

can you verify that's being called correctly by `commit_logging_object()`?

that updates the cached stats for the bucket without having to pay for re-sync (which has to read from every bucket index shard, and write the summed stats to the owner's .buckets object)

Reply:
nope. assuming the size is correct, that should cover it

Reply:
removed the explicit calls from the code